src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs

```diff
@@ -0,0 +1,201 @@
+        [Parameter(HelpMessage = "Skip confirmation message for performing the action")]
+        public SwitchParameter Force { get; set; }
+
+        /// <summary>
+        /// Defines whether the cmdlets will output the model object at the end of its execution
+        /// </summary>
+        public SwitchParameter PassThru { get; set; }
+
+        /// <summary>
+        /// Returns true if the model object that was constructed by this cmdlet should be written out
+        /// </summary>
+        protected override bool WriteResult() { return PassThru; }
+
+        /// <summary>
+        /// Get the entities from the service
+        /// </summary>
+        /// <returns>The list of entities</returns>
+        protected override IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> GetEntity()
+        {
+            return ModelAdapter.GetDatabaseLongTermRetentionBackups(
+                LocationName,
+                ServerName,
+                DatabaseName,
+                BackupName,
+                null,
+                null);
+        }
+
+        /// <summary>
+        /// No user input to apply to model
+        /// </summary>
+        /// <param name="model">Model retrieved from service</param>
+        /// <returns>The model that was passed in</returns>
+        protected override IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> ApplyUserInputToModel(
+            IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> model)
+        {
+            return model;
+        }
+
+        /// <summary>
+        /// No changes to persist to server
+        /// </summary>
+        /// <param name="entity">The output of apply user input to model</param>
+        /// <returns>The input entity</returns>
+        protected override IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> PersistChanges(
+            IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> entity)
+        {
+            ModelAdapter.RemoveDatabaseLongTermRetentionBackup(LocationName, ServerName, DatabaseName, BackupName);
+            return entity;
+        }
+
+        /// <summary>
+        /// Entry point for the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            if (InputObject != null)
+            {
+                LocationName = InputObject.LocationName;
+                ServerName = InputObject.ServerName;
+                DatabaseName = InputObject.DatabaseName;
+                BackupName = InputObject.BackupName;
+            }
+            else if (!string.IsNullOrWhiteSpace(ResourceId))
+            {
+                ResourceIdentifier identifier = new ResourceIdentifier(ResourceId);
+                BackupName = identifier.ResourceName;
+                identifier = new ResourceIdentifier(identifier.ParentResource);
+                DatabaseName = identifier.ResourceName;
+                identifier = new ResourceIdentifier(identifier.ParentResource);
```

Comment:
you don't need to set the identifier each time, do you?

Reply:
not addressed.

Reply:
Not completely necessary but the clearest way because I'm working my way up the resource id and since there are methods to get the databasename, servername, or location, this seemed like the easiest way.